File: Components/ASP.NET/CS/ASPxSchedulerDemos/App_Code/MedicsSchedulingDbModel.cs
using System;
using System.Data.Entity;
using System.Data.Entity.Infrastructure;
using System.Data.Entity.ModelConfiguration;

public partial class MedicsSchedulingDBContext : ContextBase {
    static MedicsSchedulingDBContext() {
        Database.SetInitializer<MedicsSchedulingDBContext>(null);
    }

    public MedicsSchedulingDBContext() : base("MedicsSchedulingDbConnectionString") { }

    public DbSet<MedicsSchedulingDb_MedicalAppointments> MedicalAppointments { get; set; }
    public DbSet<MedicsSchedulingDb_Medics> Medics { get; set; }

    protected override void OnModelCreating(DbModelBuilder modelBuilder) {
        modelBuilder.Configurations.Add(new MedicsSchedulingDb_MedicalAppointmentsMap());
        modelBuilder.Configurations.Add(new MedicsSchedulingDb_MedicsMap());
    }
}

public partial class MedicsSchedulingDb_MedicalAppointments {
    public int ID { get; set; }
    public Nullable<int> MedicId { get; set; }
    public string MedicIds { get; set; }
    public Nullable<int> Status { get; set; }
    public string Subject { get; set; }
    public string Description { get; set; }
    public Nullable<int> Label { get; set; }
    public Nullable<System.DateTime> StartTime { get; set; }
    public Nullable<System.DateTime> EndTime { get; set; }
    public string Location { get; set; }
    public bool AllDay { get; set; }
    public Nullable<int> EventType { get; set; }
    public string RecurrenceInfo { get; set; }
    public string ReminderInfo { get; set; }
    public string ContactInfo { get; set; }
}

public partial class MedicsSchedulingDb_Medics {
    public int ID { get; set; }
    public string Name { get; set; }
    public string DisplayName { get; set; }
    public string Department { get; set; }
    public string Phone { get; set; }
    public byte[] PhotoBytes { get; set; }
}

#region Mapping
public class MedicsSchedulingDb_MedicalAppointmentsMap : EntityTypeConfiguration<MedicsSchedulingDb_MedicalAppointments> {
    public MedicsSchedulingDb_MedicalAppointmentsMap() {
        // Primary Key
        this.HasKey(t => t.ID);

        // Properties
        this.Property(t => t.Subject)
            .HasMaxLength(50);

        this.Property(t => t.Location)
            .HasMaxLength(50);

        // Table & Column Mappings
        this.ToTable("MedicalAppointments");
        this.Property(t => t.ID).HasColumnName("ID");
        this.Property(t => t.MedicId).HasColumnName("MedicId");
        this.Property(t => t.MedicIds).HasColumnName("MedicIds");
        this.Property(t => t.Status).HasColumnName("Status");
        this.Property(t => t.Subject).HasColumnName("Subject");
        this.Property(t => t.Description).HasColumnName("Description");
        this.Property(t => t.Label).HasColumnName("Label");
        this.Property(t => t.StartTime).HasColumnName("StartTime");
        this.Property(t => t.EndTime).HasColumnName("EndTime");
        this.Property(t => t.Location).HasColumnName("Location");
        this.Property(t => t.AllDay).HasColumnName("AllDay");
        this.Property(t => t.EventType).HasColumnName("EventType");
        this.Property(t => t.RecurrenceInfo).HasColumnName("RecurrenceInfo");
        this.Property(t => t.ReminderInfo).HasColumnName("ReminderInfo");
        this.Property(t => t.ContactInfo).HasColumnName("ContactInfo");
    }
}

public class MedicsSchedulingDb_MedicsMap : EntityTypeConfiguration<MedicsSchedulingDb_Medics> {
    public MedicsSchedulingDb_MedicsMap() {
        // Primary Key
        this.HasKey(t => t.ID);

        // Properties
        this.Property(t => t.ID);

        this.Property(t => t.Name)
            .HasMaxLength(50);

        this.Property(t => t.DisplayName)
            .HasMaxLength(50);

        this.Property(t => t.Department)
            .HasMaxLength(50);

        this.Property(t => t.Phone)
            .HasMaxLength(50);

        // Table & Column Mappings
        this.ToTable("Medics");
        this.Property(t => t.ID).HasColumnName("ID");
        this.Property(t => t.Name).HasColumnName("Name");
        this.Property(t => t.DisplayName).HasColumnName("DisplayName");
        this.Property(t => t.Department).HasColumnName("Department");
        this.Property(t => t.Phone).HasColumnName("Phone");
        this.Property(t => t.PhotoBytes).HasColumnName("PhotoBytes");
    }
}
#endregion

File: Components/ASP.NET/VB/ASPxSchedulerDemos/App_Code/MedicsSchedulingDbModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data.Entity
Imports System.Data.Entity.Infrastructure
Imports System.Data.Entity.ModelConfiguration

Partial Public Class MedicsSchedulingDBContext
	Inherits ContextBase
	Shared Sub New()
		Database.SetInitializer(Of MedicsSchedulingDBContext)(Nothing)
	End Sub

	Public Sub New()
		MyBase.New("MedicsSchedulingDbConnectionString")
	End Sub

	Public Property MedicalAppointments() As DbSet(Of MedicsSchedulingDb_MedicalAppointments)
	Public Property Medics() As DbSet(Of MedicsSchedulingDb_Medics)

	Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
		modelBuilder.Configurations.Add(New MedicsSchedulingDb_MedicalAppointmentsMap())
		modelBuilder.Configurations.Add(New MedicsSchedulingDb_MedicsMap())
	End Sub
End Class

Partial Public Class MedicsSchedulingDb_MedicalAppointments
	Public Property ID() As Integer
	Public Property MedicId() As Integer?
	Public Property MedicIds() As String
	Public Property Status() As Integer?
	Public Property Subject() As String
	Public Property Description() As String
	Public Property Label() As Integer?
	Public Property StartTime() As System.DateTime?
	Public Property EndTime() As System.DateTime?
	Public Property Location() As String
	Public Property AllDay() As Boolean
	Public Property EventType() As Integer?
	Public Property RecurrenceInfo() As String
	Public Property ReminderInfo() As String
	Public Property ContactInfo() As String
End Class

Partial Public Class MedicsSchedulingDb_Medics
	Public Property ID() As Integer
	Public Property Name() As String
	Public Property DisplayName() As String
	Public Property Department() As String
	Public Property Phone() As String
	Public Property PhotoBytes() As Byte()
End Class

#Region "Mapping"
Public Class MedicsSchedulingDb_MedicalAppointmentsMap
	Inherits EntityTypeConfiguration(Of MedicsSchedulingDb_MedicalAppointments)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.ID)

		' Properties
		Me.Property(Function(t) t.Subject).HasMaxLength(50)

		Me.Property(Function(t) t.Location).HasMaxLength(50)

		' Table & Column Mappings
		Me.ToTable("MedicalAppointments")
		Me.Property(Function(t) t.ID).HasColumnName("ID")
		Me.Property(Function(t) t.MedicId).HasColumnName("MedicId")
		Me.Property(Function(t) t.MedicIds).HasColumnName("MedicIds")
		Me.Property(Function(t) t.Status).HasColumnName("Status")
		Me.Property(Function(t) t.Subject).HasColumnName("Subject")
		Me.Property(Function(t) t.Description).HasColumnName("Description")
		Me.Property(Function(t) t.Label).HasColumnName("Label")
		Me.Property(Function(t) t.StartTime).HasColumnName("StartTime")
		Me.Property(Function(t) t.EndTime).HasColumnName("EndTime")
		Me.Property(Function(t) t.Location).HasColumnName("Location")
		Me.Property(Function(t) t.AllDay).HasColumnName("AllDay")
		Me.Property(Function(t) t.EventType).HasColumnName("EventType")
		Me.Property(Function(t) t.RecurrenceInfo).HasColumnName("RecurrenceInfo")
		Me.Property(Function(t) t.ReminderInfo).HasColumnName("ReminderInfo")
		Me.Property(Function(t) t.ContactInfo).HasColumnName("ContactInfo")
	End Sub
End Class

Public Class MedicsSchedulingDb_MedicsMap
	Inherits EntityTypeConfiguration(Of MedicsSchedulingDb_Medics)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.ID)

		' Properties
		Me.Property(Function(t) t.ID)

		Me.Property(Function(t) t.Name).HasMaxLength(50)

		Me.Property(Function(t) t.DisplayName).HasMaxLength(50)

		Me.Property(Function(t) t.Department).HasMaxLength(50)

		Me.Property(Function(t) t.Phone).HasMaxLength(50)

		' Table & Column Mappings
		Me.ToTable("Medics")
		Me.Property(Function(t) t.ID).HasColumnName("ID")
		Me.Property(Function(t) t.Name).HasColumnName("Name")
		Me.Property(Function(t) t.DisplayName).HasColumnName("DisplayName")
		Me.Property(Function(t) t.Department).HasColumnName("Department")
		Me.Property(Function(t) t.Phone).HasColumnName("Phone")
		Me.Property(Function(t) t.PhotoBytes).HasColumnName("PhotoBytes")
	End Sub
End Class
#End Region

